src/vm/types/signatures.rs

```diff
@@ -680,7 +788,35 @@ impl TypeSignature {
         }
         if let SymbolicExpressionType::LiteralValue(Value::Int(buff_len)) = &type_args[0].expr {
             BufferLength::try_from(*buff_len)
-                .map(|buff_len| TypeSignature::BufferType(buff_len))
+                .map(|buff_len| SequenceType(SequenceSubtype::BufferType(buff_len)))
+        } else {
+            Err(CheckErrors::InvalidTypeDescription)
+        }
+    }
+
+    // Parses type signatures of the form:
+    // (string-utf8 10)
+    fn parse_string_utf8_type_repr(type_args: &[SymbolicExpression]) -> Result<TypeSignature> {
+        if type_args.len() != 1 {
+            return Err(CheckErrors::InvalidTypeDescription)
+        }
+        if let SymbolicExpressionType::LiteralValue(Value::Int(utf8_len)) = &type_args[0].expr {
```

Comment:
Why is this `Int`?  Shouldn't it be `UInt`?  i.e. what happens if this value is negative?  (I have the same question for `buff` as well).

Reply:
Until we implement something like #1716, the type descriptors that use lengths use "int" literals, e.g.:

```
(list 10 int)
```

as opposed to:

```
(list u10 int)
```

The `try_from` constructor should throw a CheckError if a negative integer is passed in.